library/src/main/java/com/leinardi/android/speeddial/UiUtils.java

```diff
@@ -166,26 +166,73 @@ public static void rotateForward(View view, boolean animate) {
-                .setInterpolator(new AccelerateDecelerateInterpolator())
+                .setInterpolator(new FastOutSlowInInterpolator())
                 .start();
     }
 
-    public static Drawable getRotateDrawable(Context context, @DrawableRes int drawableRes) {
-        final Drawable drawable = AppCompatResources.getDrawable(context, drawableRes);
-        Canvas canvas = new Canvas();
-        Bitmap bitmap = Bitmap.createBitmap(drawable.getIntrinsicWidth(), drawable.getIntrinsicHeight(), Bitmap
-                .Config.ARGB_8888);
-        canvas.setBitmap(bitmap);
-        canvas.rotate(-ROTATION_ANGLE, drawable.getIntrinsicWidth() / 2, drawable.getIntrinsicHeight() / 2);
-        drawable.setBounds(0, 0, drawable.getIntrinsicWidth(), drawable.getIntrinsicHeight());
-        drawable.draw(canvas);
-        return new BitmapDrawable(context.getResources(), bitmap);
+    public static Drawable getRotateDrawable(final Drawable drawable, final float angle) {
+        if (angle == 0) {
+            return drawable;
+        }
+        final Drawable[] drawables = {drawable};
+        return new LayerDrawable(drawables) {
+            @Override
+            public void draw(final Canvas canvas) {
+                canvas.save();
+                canvas.rotate(angle, drawable.getBounds().width() / 2, drawable.getBounds().height() / 2);
+                super.draw(canvas);
+                canvas.restore();
+            }
+        };
+    }
+
+    /**
+     * Creates a {@link Bitmap} from a {@link Drawable}.
+     */
+    @Nullable
+    public static Bitmap getBitmapFromDrawable(@Nullable Drawable drawable) {
+        if (drawable == null) {
+            return null;
+        } else {
+            Bitmap bitmap;
+            if (drawable instanceof BitmapDrawable) {
+                BitmapDrawable bitmapDrawable = (BitmapDrawable) drawable;
+                if (bitmapDrawable.getBitmap() != null) {
+                    return bitmapDrawable.getBitmap();
+                }
+            }
+
+            if (drawable.getIntrinsicWidth() <= 0 || drawable.getIntrinsicHeight() <= 0) {
+                // Single color bitmap will be created of 1x1 pixel
+                bitmap = Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888);
+            } else {
+                bitmap = Bitmap.createBitmap(drawable.getIntrinsicWidth(), drawable.getIntrinsicHeight(), Bitmap
+                        .Config.ARGB_8888);
+            }
+
+            Canvas canvas = new Canvas(bitmap);
+            drawable.setBounds(0, 0, canvas.getWidth(), canvas.getHeight());
+            drawable.draw(canvas);
+            return bitmap;
+        }
+    }
+
+    /**
+     * Creates a {@link Drawable} from a {@link Bitmap}.
+     */
+    @Nullable
+    public static Drawable getDrawableFromBitmap(@Nullable Bitmap bitmap) {
+        if (bitmap == null) {
+            return null;
+        } else {
+            return new BitmapDrawable(bitmap);
```

Comment:
**Reporter**: Error Prone
**Rule**: deprecation
**Severity**: WARN
**File**: library/src/main/java/com/leinardi/android/speeddial/UiUtils.java L235

BitmapDrawable(Bitmap) in BitmapDrawable has been deprecated

return new BitmapDrawable(bitmap);^

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<-1252870220>*